### PR TITLE
1050 (re)populate OpenSearch

### PR DESCRIPTION
**Pull request by leite08**
Ref: https://github.com/metriport/metriport-internal/issues/1050

### Dependencies

- upstream: https://github.com/metriport/metriport/pull/906

### Description

Went with the script/utility approach to avoid having an endpoint that can generate a load large enough to make systems unresponsive.

On the flip side, running this script is dangerous as it relies on (1) manual setup of a `.env` file and (2) a local environment.

For (1), we can shoulder check the env setup before running it. For (2), the worst that could happen is stopping the process midway through, which is non-destructive.

Changes presented by this PR:

- add script/utilily to repopulate OpenSearch (OS)
- beef up the prod open search instance 
- update docs w/ OS search syntax
- ingestion to upsert instead of insert
- fix ingestion error handling
- move HAPI client to core
- FHIR URL required on API
- fix tester lambda entry

PS.: there are more commits than I'd expect to see there, I'm pretty sure that's because of some issues I had while rebasing from develop into upstream and from there to this branch; but what matters is the diff and I checked and all the changes on this PR are expected.

### Release Plan

- ⚠️ Run this off hours
- [ ] merge this
- [ ] run the `re-populate` script, w/ `.env` pointing to `sandbox` and `production`
   - ingesting files on `staging` took 5 seconds to ingest 43 documents
   - we have more than 10,000 documents in `prod`, so it will take a while

### CODE REVIEW

File: packages/api/src/external/commonwell/document/document-query.ts [hunk 1]
```diff
@@ -597,21 +597,16 @@ export async function downloadDocsAndUpsertFHIR({
...
                 errorReported = true;
                 throw error;
               }),
-              ingestIntoSearchEngine(patient, FHIRDocRef, file, requestId).catch(error => {
-                const context = "ingestIntoSearchEngine";
-                reportFHIRError({ patientId: patient.id, doc, error, context, log });
-                errorReported = true;
-                throw error;
-              }),
+              ingestIntoSearchEngine(patient, FHIRDocRef, file, requestId, log),
```
Comment (leite08) [inline packages/api/src/external/commonwell/document/document-query.ts:607]: Intentionally making the search ingestion to fail gracefully so it doesn't impact the download progress for cxs. The function has its own error handling code notifying Sentry.

File: packages/api/src/shared/config.ts [hunk 2]
```diff
@@ -140,13 +140,8 @@ export class Config {
...
 
-  static getFHIRServerUrl(): string | undefined {
-    return getEnvVar("FHIR_SERVER_URL");
-  }
-  static getFHIRServerUrlOrFail(): string {
-    const url = Config.getFHIRServerUrl();
-    if (!url) throw new Error(`Missing FHIR_SERVER_URL env var, env: ${Config.getEnvType()}`);
-    return url;
+  static getFHIRServerUrl(): string {
+    return getEnvVarOrFail("FHIR_SERVER_URL");
```
Comment (leite08) [inline packages/api/src/shared/config.ts:144]: All envs set this env var, so could make it required and the code cleaner.

File: docs/medical-api/api-reference/document/list-documents.mdx [hunk 3]
```diff
@@ -27,7 +27,21 @@ query, see the [Start Document Query endpoint](/medical-api/api-reference/docume
 </ParamField>
 
 <ParamField query="content" type="string" optional>
-  Value to search on the document reference (partial match and case insentitive, minimum 3 chars).
+  Value to search on the document reference (minimum 3 chars). For more details see the "search syntax" section
+  below.
+  
+  <Expandable title="Search syntax">
```
Comment (Goncharo) [inline docs/medical-api/api-reference/document/list-documents.mdx:33]: nit: with search syntax with multiple rules, usually practical examples with a text description are useful

**Review by Goncharo — APPROVED** [review]
one nit to consider, otherwise :shipit:

**Review by Goncharo — APPROVED** [review]

**Pull request merged**